Gems/LmbrCentral/Code/Source/Bundling/BundlingSystemComponent.cpp

```diff
@@ -114,15 +126,130 @@ namespace LmbrCentral
+
+        AZStd::lock_guard<AZStd::mutex> openBundleLock(m_bundleModeMutex);
+
+        // NOTE:  'bundlePath' must produce an identical result to paks discovered below in "LoadBundles"
+        AZStd::string bundlePath;
+        AZ::StringFunc::Path::Join(bundleRoot, szPath, bundlePath);
+
+        bool pakOpened = false;
+        for (const auto& openedBundle : m_bundleModeBundles)
+        {
+            if (openedBundle == bundlePath)
+            {
+                pakOpened = true;
+                break;
+            }
+        }
+
+        if (pakOpened)
+        {
+            if (reload)
+            {
+                if (archive->IsPackOpen(AZStd::string_view(szPath)))
+                {
+                    if (!archive->ClosePack(szPath))
+                    {
+                        AZ_Warning("BundlingSystem", false, "Failed to close bundle %s", szPath);
+                    }
+                }
+
+                m_bundleModeBundles.erase(AZStd::remove(m_bundleModeBundles.begin(), m_bundleModeBundles.end(), bundlePath.c_str()), m_bundleModeBundles.end());
+            }
+            else
+            {
+                return;
+            }
+        }
+
+        if (archive->OpenPack(bundleRoot, szPath))
+        {
+            AZ_TracePrintf("BundlingSystem", "Loaded bundle %s\n", bundlePath.c_str());
+            m_bundleModeBundles.emplace_back(AZStd::move(bundlePath));
+        }
+        else
+        {
+            AZ_TracePrintf("BundlingSystem", "Failed to load %s\n", bundlePath.c_str());
+        }
+    }
+
+    AZStd::vector<AZStd::string> BundlingSystemComponent::GenerateBundlesToLoadList(const char* bundleFolder, const char* bundleExtension)
+    {
+        AZStd::vector<AZStd::string> bundleToLoadList;
+        AZStd::vector<AZStd::string> bundleList = GetBundleList(bundleFolder, bundleExtension);
+        AZ_TracePrintf("BundlingSystem", "Loading bundles from %s of type %s\n", bundleFolder, bundleExtension);
+        if (!bundleList.size())
+        {
+            AZ_Warning("BundlingSystem", false, "Failed to locate bundles of type %s in folder %s", bundleExtension, bundleFolder);
+            return bundleToLoadList;
+        }
+
+        AZStd::lock_guard<AZStd::mutex> openBundleLock(m_bundleModeMutex);
+        for (const auto& thisBundle : bundleList)
+        {
+            AZStd::string bundlePath;
+            AZ::StringFunc::Path::Join(bundleRoot, thisBundle.c_str(), bundlePath);
+
+            bool pakOpened = false;
+            for (const auto& openedBundle : m_bundleModeBundles)
+            {
+                // Comparing openedBundle with bundlePath instead of 'thisBundle' since the path will get normalized
+                if (openedBundle == bundlePath)
+                {
+                    pakOpened = true;
+                    break;
+                }
+            }
+
+            if (!pakOpened)
+                bundleToLoadList.push_back(thisBundle);
+        }
+
+#if defined(CARBONATED)
```

Comment:
This is CARBONATED in CARBONATED, redundant.

Reply:
Updated